src/Functions/parseDateTime.cpp

```diff
@@ -0,0 +1,1766 @@
+           39447, 39812, 40177, 40543, 40908, 41273, 41638, 42004, 42369, 42734, 43099, 43465, 43830, 44195, 44560, 44926, 45291, 45656,
+           46021, 46387, 46752, 47117, 47482, 47847, 48212, 48577, 48942, 49308, 49673};
+
+    struct DateTime
+    {
+        /// If both week_date_format and week_date_format is false, date is composed of year, month and day
+        Int32 year = 1970; /// year, range [1970, 2106]
+        Int32 month = 1; /// month of year, range [1, 12]
+        Int32 day = 1; /// day of month, range [1, 31]
+
+        Int32 week = 1; /// ISO week of year, range [1, 53]
+        Int32 day_of_week = 1; /// day of week, range [1, 7], 1 represents Monday, 2 represents Tuesday...
+        bool week_date_format
+            = false; /// If true, date is composed of week year(reuse year), week of year(use week) and day of week(use day_of_week)
+
+        Int32 day_of_year = 1; /// day of year, range [1, 366]
+        bool day_of_year_format = false; /// If true, date is composed of year(reuse year), day of year(use day_of_year)
+
+        bool century_format = false; /// If true, year is calculated from century, range [19, 21]
+
+        bool is_year_of_era = false; /// If true, year is calculated from era and year of era, the latter cannot be zero or negative.
+        bool has_year = false; /// Whether year was explicitly specified.
+
+        /// If is_clock_hour = true, is_hour_of_half_day = true, hour's range is [1, 12]
+        /// If is_clock_hour = true, is_hour_of_half_day = false, hour's range is [1, 24]
+        /// If is_clock_hour = false, is_hour_of_half_day = true, hour's range is [0, 11]
+        /// If is_clock_hour = false, is_hour_of_half_day = false, hour's range is [0, 23]
+        Int32 hour = 0;
+        Int32 minute = 0; /// range [0, 59]
+        Int32 second = 0; /// range [0, 59]
+
+        bool is_am = true; /// AM -> true, PM -> false
+        bool is_clock_hour = false; /// Whether the hour is clockhour
+        bool is_hour_of_half_day = false; /// Whether the hour is of half day
+
+        bool has_time_zone_offset = false; /// If true, time zone offset if explicitly specified.
+        Int64 time_zone_offset = 0; /// Offset in seconds between current timezone to UTC.
+
+        void reset()
+        {
+            year = 1970;
+            month = 1;
+            day = 1;
+
+            week = 1;
+            day_of_week = 1;
+            week_date_format = false;
+
+            day_of_year = 1;
+            day_of_year_format = false;
+
+            century_format = false;
+
+            is_year_of_era = false;
+            has_year = false;
+
+            hour = 0;
+            minute = 0;
+            second = 0;
+
+            is_am = true;
+            is_clock_hour = false;
+            is_hour_of_half_day = false;
+
+            has_time_zone_offset = false;
+            time_zone_offset = 0;
+        }
+
+        void setCentury(Int32 century)
```

Comment:
Could we please order below functions in class `DateTime` descendingly in their time span, e.g. setCentury, set Year, setMonth etc.? (right now, the order is a bit messy)